src/i18n-module.php

```diff
@@ -111,23 +111,27 @@ class Yoast_I18n_v2 {
 	/**
 	 * Class constructor
 	 *
-	 * @param array $args Contains the settings for the class.
+	 * @param array $args                   Contains the settings for the class.
+	 * @param bool $show_translation_box    Whether the translation box should be shown.
 	 */
-	public function __construct( $args ) {
-		if ( ! is_admin() ) {
-			return;
-		}
+	public function __construct( $args, $show_translation_box = true ) {
+		if( $this->is_admin_in_other_language() ) {
+			$this->init( $args );
 
-		$this->locale = $this->get_admin_locale();
-		if ( 'en_US' === $this->locale ) {
-			return;
+			if ( $show_translation_box ) {
```

Comment:
You removed the `hide_promo` check. This check is looking to the dismiss state of the box. 

`if ( $show_translation_box && ! $this->hide_promo() ) {`